### PR TITLE
CORTX-29435: Add GitHub Pull Request template

### PR DESCRIPTION
**Pull request by keithpine**
This change adds a Pull Request template in order to improve and streamline PR submissions.

Signed-off-by: Keith Pine <keith.pine@seagate.com>

-----
[View rendered .github/PULL_REQUEST_TEMPLATE.md](https://github.com/keithpine/cortx-k8s/blob/CORTX-29435_pr-template/.github/PULL_REQUEST_TEMPLATE.md)

### CODE REVIEW

**Comment by keithpine** [comment]
The Codacy checks are invalid.

**Review by walterlopatka — APPROVED** [review]
[3:55 PM] Walter Lopatka
Looks good to me.  The only thing not listed there that I think could be helpful in a PR is how the testing was done.  I'm not ready to make that a requirement, though.  I'm happy with it as it is, and I guess we will learn over the next couple of sprints what needs to be changed.

**Review by walterlopatka — APPROVED** [review]
I like the "how was this tested?" section. Nicely done.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+completely. Unless instructed otherwise, do not delete any sections.
+-->
+## Description
+<!--
+Describe what this change does and the motivation behind it. Why is it required? What problems does
+it solve?
+-->
+
+## Breaking change
+<!--
```
Comment (osowski) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: Would we want to add something in this section along the lines of "Is a breaking change required? Could a breaking change be avoided by tolerant deprecation instead?" or something similar.

Comment (osowski) [reply .github/PULL_REQUEST_TEMPLATE.md:12]: Perfect! 🎉

**Pull request merged**